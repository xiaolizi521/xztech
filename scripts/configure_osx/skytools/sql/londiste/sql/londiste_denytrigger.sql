
create table denytest ( val integer);
insert into denytest values (1);
create trigger xdeny after insert or update or delete
on denytest for each row execute procedure londiste.deny_trigger();

insert into denytest values (2);
update denytest set val = 2;
delete from denytest;

select londiste.disable_deny_trigger(true);
update denytest set val = 2;
select londiste.disable_deny_trigger(true);
update denytest set val = 2;
select londiste.disable_deny_trigger(false);
update denytest set val = 2;
select londiste.disable_deny_trigger(false);
update denytest set val = 2;

