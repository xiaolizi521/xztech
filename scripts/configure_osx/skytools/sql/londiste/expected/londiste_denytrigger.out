create table denytest ( val integer);
insert into denytest values (1);
create trigger xdeny after insert or update or delete
on denytest for each row execute procedure londiste.deny_trigger();
insert into denytest values (2);
ERROR:  ('Changes no allowed on this table',)
update denytest set val = 2;
ERROR:  ('Changes no allowed on this table',)
delete from denytest;
ERROR:  ('Changes no allowed on this table',)
select londiste.disable_deny_trigger(true);
 disable_deny_trigger 
----------------------
 t
(1 row)

update denytest set val = 2;
select londiste.disable_deny_trigger(true);
 disable_deny_trigger 
----------------------
 t
(1 row)

update denytest set val = 2;
select londiste.disable_deny_trigger(false);
 disable_deny_trigger 
----------------------
 f
(1 row)

update denytest set val = 2;
ERROR:  ('Changes no allowed on this table',)
select londiste.disable_deny_trigger(false);
 disable_deny_trigger 
----------------------
 f
(1 row)

update denytest set val = 2;
ERROR:  ('Changes no allowed on this table',)
